### PR TITLE
Make MonadWide up/down focus navigation behave like MonadTall left/right

### PR DESCRIPTION
**Pull request by holocronweaver**
MonadTall exposes left/right commands to focus the closest window to the left/right, allowing easy navigation between columns. MonadWide had no similar override for up/down focus commands, instead it inherited the next/previous = down/up overrides of MonadTall, making directional navigation between rows more cumbersome by requiring cycling amongst all windows. The result is inconsistent window navigation between MonadTall and MonadWide.

This change retains the original navigation commands and behavior of MonadTall (including left/right), while adding a symmetric up/down command to MonadWide.

---

Besides manually testing the Qtile directional focus navigation behavior for both MonadTall and Wide layouts, I added unit tests, and have verified the full test suite passes locally (except for one related to migration that seems unrelated to my changes). I also verified that my unit tests for MonadTall pass without my layout code changes (implying no behavior change), while those for MonadWide fail (implying I fixed the inconsistency).

### CODE REVIEW

File: libqtile/layout/xmonad.py [hunk 1]
```diff
@@ -858,6 +858,16 @@ def right(self):
...
+    @expose_command()
+    def up(self):
+        """Focus on the closest window above the current window"""
+        self.previous()
+
+    @expose_command()
+    def down(self):
+        """Focus on the closest window below the current window"""
+        self.next()
+
```
Comment (elParaguayo) [inline libqtile/layout/xmonad.py:870]: You're probably correct, but why is this needed. The previous decorators should have had the same effect as this. Does it cause issues for `MonadWide` on inheritance?

Comment (holocronweaver) [reply libqtile/layout/xmonad.py:870]: Exactly, the decorators caused issues with `MonadWide` on inheritance. I tried overriding in `MonadWide` and leaving the decorators as is, but the overrides were ignored and the decorators in `MonadTall` took precedence (confirmed by adding some log lines). Avoiding decorators allowed inheritance and overrides to work as usual.
